docs/azure/dependency-injection.md

```diff
@@ -0,0 +1,233 @@
+  }
+}
+```
+
+## Store configuration separately from code
+
+Recall in the [Register client](#register-client) section above, you explicitly specify the `keyVaultUrl` and `storageUrl`. This could cause problems when you run code against different environments during development and production. The ASP.NET Core team suggests [storing such configurations in environment dependent JSON files](/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#default-configuration). Thus, you can have an `appsettings.Development.json` file with one set of settings and an `appsettings.Production.json` with another set of configurations. The format of the file is:
+
+```json
+{
+  "AzureDefaults": {
+    "Diagnostics": {
+      "IsTelemetryDisabled": false,
+      "IsLoggingContentEnabled": true
+    },
+    "Retry": {
+      "MaxRetries": 3,
+      "Mode": "Exponential"
+    }
+  },
+  "KeyVault": {
+    "VaultUri": "https://mykeyvault.vault.azure.net"
+  },
+  "Storage": {
+    "ServiceUri": "https://mydemoaccount.storage.windows.net"
+  }
+}
+```
+
+You can add any options from the [ClientOptions](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/ClientOptions.cs) into the `AzureDefaults` section. One of the options is the retry policy. See the section [Configure a new try policy](#configure-a-new-retry-policy).
+
+Since the `Configuration` object is injected from the host and stored inside the `Startup` constructor, you can do the following:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+  services.AddAzureClients(builder =>
+  {
+    // Add a KeyVault client
+    builder.AddSecretClient(Configuration.GetSection("KeyVault"));
+
+    // Add a storage account client
+    builder.AddBlobServiceClient(Configuration.GetSection("Storage"));
+
+    // Use DefaultAzureCredential by default
+    builder.UseCredential(new DefaultAzureCredential());
+
+    // Set up any default settings
+    builder.ConfigureDefaults(Configuration.GetSection("AzureDefaults"));
+  });
+
+  services.AddControllers();
+}
+```
+
+## Configure multiple service clients with different names
+
+Say you have two storage accounts – one for private information and one for public information. Your application transfers data from the public to private storage account after some operation. You need to have two storage service clients. To set this up in `Startup.ConfigureServices`:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+  services.AddAzureClients(builder =>
+  {
+    builder.AddBlobServiceClient(Configuration.GetSection("PublicStorage"));
+    builder.AddBlobServiceClient(Configuration.GetSection("PrivateStorage"))
+      .WithName("PrivateStorage");
+  });
+}
+```
+
+In your controllers, you can access the named service clients using the `IAzureClientFactory`:
+
+```csharp
+public class HomeControllers : Controller
+{
+  private BlobServiceClient publicStorage, privateStorage;
```

Comment:
```suggestion
  private readonly BlobServiceClient _publicStorage;
  private readonly BlobServiceClient _privateStorage;
```

These should be `readonly`, on their own line, and follow the proper naming conventions, leading prefix of `_` for private fields.